django/db/models/base.py

```diff
@@ -519,11 +519,14 @@ def __hash__(self):
         return hash(self.pk)
 
     def __reduce__(self):
-        data = self.__dict__
+        data = self.__getstate__()
         data[DJANGO_VERSION_PICKLE_KEY] = get_version()
         class_id = self._meta.app_label, self._meta.object_name
         return model_unpickle, (class_id,), data
 
+    def __getstate__(self):
+        return self.__dict__.copy()
```

Comment:
In one of my project we wrote some mixins for models that added some required default behaviour. In some moment several mixins added new instance attributes that mustn't be stored and used only in runtime (we're using redis for caching and pickle for serializing). 

If copy must be done on a subclass level then it is harder to understand was copy created already or not, which can lead to extra copying or tricky ways of remembering that copy was already done by some other class.